l3kernel/doc/source3body.tex

```diff
@@ -335,6 +335,14 @@ \section{Naming functions and variables}
     lists (with access at both ends) and stacks.
 \end{description}
 
+\subsection{Behavior of \texttt{c}-type expansion for undefined arguments}
+
+When \texttt{c}-type expansion is applied to with and argument that is
+undefined, the result is a token equal to |\scan_stop:| (|\relax|). This will
+likely lead to low-level errors if applied to variables. As such, the
+programmer should ensure that \texttt{c}-type expansion is only applied where
+resulting variable names will definitely exist.
```

Comment:
And now we have "variable names": 
which seems to imply that these names vary!! 

So again: what is "a variable"?

Are the variable and its name not the same thing?
Can the variable exist but not its name, or vice versa?

Reply:
I've tried to make it clear what I was getting at